### ISSUE TITLE
Kafka threads are still running when server shutdown is initiated

### ISSUE THREAD

**Issue by arsalans** [reactor/reactor-kafka#161]
When we shutdown our JBOSS application server, it hangs. We created a ticket with Redhat and provided them with thread dumps when JVM is hung. They analyzed the dumps and said that that jboss threads are waiting for Kafka threads to shutdown. Here is the response from them:

```
From the last thread dumps, the graceful shutdown initiated, but it seems it is still waiting for some service to stop:

"main" #1 prio=5 os_prio=0 tid=0x00007f55ac009800 nid=0x43a3 in Object.wait() [0x00007f55b5648000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at org.jboss.as.server.GracefulShutdownService.awaitSuspend(GracefulShutdownService.java:86)
	- locked <0x00000005c0881bb8> (a java.lang.Object)
	at org.jboss.as.server.DomainServerMain.main(DomainServerMain.java:170)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.modules.Module.run(Module.java:329)
	at org.jboss.modules.Main.main(Main.java:507)

The http threads are idle, there are no running requests during the shutdown.

But there are some org.springframework.kafka threads running, and it seems that they are still running and do not initiate the shutdown:

"org.springframework.kafka.KafkaListenerEndpointContainer#0-0-C-1" #334 prio=5 os_prio=0 tid=0x00007f5548b08800 nid=0x454d runnable [0x00007f55269e0000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:93)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000005ce2925c8> (a sun.nio.ch.Util$3)
	- locked <0x00000005ce2925b8> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000005ce286d48> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at org.apache.kafka.common.network.Selector.select(Selector.java:794)
	at org.apache.kafka.common.network.Selector.poll(Selector.java:467)
	at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:539)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:262)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:233)
	at org.apache.kafka.clients.consumer.KafkaConsumer.pollForFetches(KafkaConsumer.java:1281)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1225)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1159)
	at brave.kafka.clients.TracingConsumer.poll(TracingConsumer.java:89)
	at brave.kafka.clients.TracingConsumer.poll(TracingConsumer.java:83)
	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.doPoll(KafkaMessageListenerContainer.java:1071)
	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.pollAndInvoke(KafkaMessageListenerContainer.java:1027)
	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.run(KafkaMessageListenerContainer.java:958)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)

ThreadPoolTaskScheduler from org.springframework is also still running: 

"ThreadPoolTaskScheduler-1" #293 prio=5 os_prio=0 tid=0x00007f556548c000 nid=0x4513 waiting on condition [0x00007f5535849000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000005cada40d8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

These classes are not managed by JBoss as they are not provided with JBoss. 
I would suggest to contact the development team or org.springframework.kafka.* support as these classes are not shutting down.
```

## Expected Behavior
Jboss should shutdown gracefully

## Actual Behavior
Jboss hangs upon shutdown

## Steps to Reproduce
Using Kafka reactor as hotflux and initiate Jboss server shutdown

## Your Environment
Spring Boot 2.2.8.RELEASE
Red Hat JBoss EAP 7.0
reactor-kafka 1.2.2.RELEASE
Java 8
Linux 3.10

**Comment by bsideup** [comment]
Duplicate of https://github.com/reactor/reactor-kafka/issues/159
